test/unit/org/apache/cassandra/net/AsyncStreamingInputPlusTest.java

```diff
@@ -161,6 +129,61 @@ public void available_ClosedButWithBytes()
         Assert.assertEquals(size, inputPlus.unsafeAvailable());
     }
 
+    @Test
+    public void rebufferAndCloseToleratesInterruption() throws InterruptedException
+    {
+        ByteBuf beforeInterrupt = channel.alloc().heapBuffer(1024);
+        beforeInterrupt.writeCharSequence("BEFORE", Charset.forName("us-ascii"));
+        final int beforeInterruptBytes = beforeInterrupt.readableBytes();
+        ByteBuf afterInterrupt = channel.alloc().heapBuffer(1024);
+        afterInterrupt.writeCharSequence("AFTER", Charset.forName("us-ascii"));
+        final int afterInterruptBytes = afterInterrupt.readableBytes();
+
+        inputPlus = new AsyncStreamingInputPlus(channel);
+        Thread consumer = new Thread(() -> {
+            try
+            {
+                byte[] buffer = new byte[beforeInterruptBytes + afterInterruptBytes];
+                inputPlus.read(buffer, 0, beforeInterruptBytes + afterInterruptBytes);
```

Comment:
Would it be worth it to throw a quick check for the number of bytes read?

```
int bytesRead = inputPlus.read(buffer, 0, beforeInterruptBytes + afterInterruptBytes);
assertEquals(buffer.length, bytesRead);
```

Reply:
Can't hurt.